packages/core/core/src/services/document-service/internationalization.ts

```diff
@@ -36,22 +36,7 @@ const localeToLookup: Transform = (contentType, params) => {
   }
 
   if (params.locale) {
-    return assoc(['lookup', 'locale'], params.locale, params);
-  }
-
-  return params;
-};
-
-/**
- * Add locale lookup query to the params
- */
-const multiLocaleToLookup: Transform = (contentType, params) => {
-  if (!strapi.plugin('i18n').service('content-types').isLocalizedContentType(contentType)) {
-    return params;
-  }
-
-  if (params.locale) {
-    if (params.locale === '*') {
+    if (typeof params.locale === 'string' && params.locale === '*') {
```

Comment:
won't this change allow doing a find many on multiple locales at the same time?
If we want to prevent that I believe we should keep the localetoLookup & multiLocaletoLookup